src/policy/immix/block.rs

```diff
@@ -248,8 +248,13 @@ impl Block {
                         holes += 1;
                     }
 
-                    #[cfg(feature = "immix_zero_on_release")]
-                    crate::util::memory::zero(line.start(), Line::BYTES);
+                    // Cheeky trick to encode the released line into the poisoned value
+                    #[cfg(feature = "poison_on_release")]
+                    crate::util::memory::set_pattern(
+                        line.start(),
+                        (0xdeadbeef ^ line.start().as_usize()) & !0xff,
```

Comment:
Why poison as a pattern rather than a `u8` value as other spaces? What is the pattern here? How do we supposed to use the pattern?

Reply:
This removes the existing code for `immix_zero_on_release`. So immix will not zero lines even if `immix_zero_on_release` is enabled.

You should have an assertion to make sure that `immix_zero_on_release` and `poison_on_release` should not co-exist, as they conflict with each other for Immix space, and you should zero/poison the memory based on which feature is enabled.

Reply:
This feature supersedes zero on release. The pattern here is helpful since it encodes the line that was released into the pattern. So you can XOR with `0xdeadbeef` to get the exact line that was released and where the object comes from. It was useful to me while debugging ART.

Reply:
`immix_zero_on_release` was a debugging feature to catch untraced objects. But using `0x0` can actually make it harder to debug (in my and others' experience) since you don't know if it's a null pointer or something that was released early. So previously some of the students and I used an obvious nonsense value like `0xab` as the pattern. This is refinement of the same since it encodes the line into the pattern.

Reply:
If you think "poison_on_release" is better than "zero_on_release", just remove the `immix_zero_on_release` feature, and make sure that `poison_on_release` can be a replacement for `immix_zero_on_release`.

Reply:
> So you can XOR with 0xdeadbeef to get the exact line that was released and where the object comes from. 

I am not following here. Can you get the line by aligning down from the faulty address? Why is XOR needed?

Reply:
The problem isn't the line itself but what object was referring to it. In my case I did not have a debugger available because crashes were happening at Android boot time. So I had to be creative to debug memory corruption errors (hence the `dump_ram_around_address` function as well).

So consider the case that a line has been overwritten when an object inside it was live.

Object = `0x12000040`
Line = `0x12000000`
Encoded pattern = `0xCCADBE00`

An object referring to `0x12000040` would maybe load a field at an offset `0x8`. So it loads the address `0x12000048`. If we wrote just `0x0`s into the line then we would get a SIGSEGV but then no way to figure out what address was loaded. But if we wrote `0xCCADBE00` to it then the SIGSEGV output would list say `0xCCADBE00` and then we can XOR with `0xDEADBE00` and get back `0x12000000` which was the line that was overwritten.

Reply:
This is I guess unnecessary in the general case if you have a debugger, but I definitely found it useful.

Reply:
I see. Without the explanation above, I wouldn't know why you encode the value in that way, and how I am supposed to use the encoded value. Those explanation should go into the comments and the docs in the code so others won't have the same question as I did when they encounter the code and when they use this feature.

Besides, I suggest separating it into a different features:
* `poison_on_release`: this just generally poisons released memory (or dead objects) with an identifiable value.
* `immix_poison_with_line_address_on_release` (or something like this): it poisons the memory with an encoded line address.